--- kdebase-4.4.3-orig/apps/CMakeLists.txt	2010-04-29 21:54:14.000000000 +0200
+++ kdebase-4.4.3/apps/CMakeLists.txt	2010-05-28 16:31:54.683232404 +0200
@@ -19,27 +19,10 @@ add_definitions (${QT_DEFINITIONS} ${KDE
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/lib/konq ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
 
 add_subdirectory( lib )
-add_subdirectory( dolphin )
-add_subdirectory( kwrite )
 add_subdirectory( kdialog )
-add_subdirectory( keditbookmarks )
-add_subdirectory( konqueror )
-add_subdirectory( kfind )
-add_subdirectory( kappfinder )
-add_subdirectory( plasma )
 
-if ( Q_WS_MAC )
-  add_subdirectory( konsole )
-  add_subdirectory( kdepasswd )
-endif ( Q_WS_MAC )
-
-if ( Q_WS_X11 )
-  add_subdirectory( kinfocenter )
-  add_subdirectory( nsplugins )
-  add_subdirectory( konsole )
-  add_subdirectory( kdepasswd )
-endif ( Q_WS_X11 )
-macro_optional_add_subdirectory( doc )
+add_subdirectory( konsole )
+add_subdirectory( kdepasswd )
 
 if(CMAKE_SOURCE_DIR STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
     macro_display_feature_log()
--- kdebase-4.4.3/apps/CMakeLists.txt.orig	2010-05-30 14:01:43.622302829 +0200
+++ kdebase-4.4.3/apps/CMakeLists.txt	2010-05-30 14:03:21.276226877 +0200
@@ -6,7 +6,6 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SO
 #search packages used by KDE
 set(KDE_MIN_VERSION "4.4.3")
 find_package(KDE4 4.4.3 REQUIRED)
-find_package(Strigi REQUIRED)
 find_package(ZLIB REQUIRED)
 include (KDE4Defaults)
 include (MacroLibrary)
