diff -ru kdegraphics-4.4.5.orig/CMakeLists.txt kdegraphics-4.4.5/CMakeLists.txt
--- kdegraphics-4.4.5.orig/CMakeLists.txt	2010-01-06 17:59:27.000000000 +0100
+++ kdegraphics-4.4.5/CMakeLists.txt	2012-07-24 02:37:06.753970752 +0200
@@ -12,9 +12,6 @@
 find_package( Expat )
 endif( WIN32 )
 
-#require by libs/libkdcraw, gwenview, and okular
-find_package(JPEG REQUIRED)
-
 macro_optional_find_package(QImageBlitz)
 macro_log_feature(QIMAGEBLITZ_FOUND "QImageBlitz" "An image effects library" "http://sourceforge.net/projects/qimageblitz" FALSE "kdesupport" "Required to build kolourpaint and okular.")
 
@@ -29,22 +26,11 @@
   macro_log_feature(X11_xf86vmode_FOUND "X11 Video Mode Extension (xf86vm)" "An X11 extension providing access to detailed timings of video modes currently in use, and a means to modify them" "http://www.x.org" FALSE "" "X11 xvidmode extensions are needed to build kgamma.")
 endif(X11_FOUND)
 
-# Preview acess was added in Exiv2 0.18
-# IMPORTANT : Several fixes done in 0.19. Set as default version when KDE 4.4 will be released
-set(EXIV2_MIN_VERSION "0.18") 
-macro_optional_find_package(Exiv2)
-macro_log_feature(EXIV2_FOUND "Exiv2" "A library to access image metadata" "http://www.exiv2.org" FALSE ${EXIV2_MIN_VERSION} "Required to build libkexiv2 and gwenview.")
-
-macro_optional_find_package(JPEG)
-macro_log_feature(JPEG_FOUND "JPEG" "A library for reading and writing JPEG image files" "http://www.ijg.org" FALSE "" "Required to build gwenview and also provides support fof PalmDB documents in okular.")
-
 macro_optional_find_package(Kipi)
 macro_log_feature(KIPI_FOUND "KIPI" "A library for adding kipi support" "http://www.kipi-plugins.org" FALSE "" "Required to build gwenview with kipi support.")
 
 macro_optional_find_package(Kdcraw)
 #do not macro_log_feature() Kdcraw
-macro_optional_find_package(Kexiv2)
-#do not macro_log_feature() Kexiv2
 
 add_definitions (${QT_DEFINITIONS} ${QT_QTDBUS_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories (${QDBUS_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
@@ -53,13 +39,11 @@
 add_subdirectory(cmake)
 add_subdirectory(libs)
 macro_optional_add_subdirectory(strigi-analyzer)
-macro_optional_add_subdirectory(thumbnailers)
 macro_optional_add_subdirectory(kcolorchooser)
 macro_optional_add_subdirectory(kruler)
 macro_optional_add_subdirectory(svgpart)
 if(QIMAGEBLITZ_FOUND)
   macro_optional_add_subdirectory(kolourpaint)
-  macro_optional_add_subdirectory(okular)
 endif(QIMAGEBLITZ_FOUND)
 
 if(Q_WS_X11)
@@ -77,9 +61,5 @@
    macro_optional_add_subdirectory(ksaneplugin)
 endif (SANE_FOUND)
 
-if (EXIV2_FOUND)
-   macro_optional_add_subdirectory(gwenview)
-endif (EXIV2_FOUND)
-
 macro_optional_add_subdirectory(doc)
 macro_display_feature_log()
diff -ru kdegraphics-4.4.5.orig/libs/CMakeLists.txt kdegraphics-4.4.5/libs/CMakeLists.txt
--- kdegraphics-4.4.5.orig/libs/CMakeLists.txt	2009-10-15 18:32:02.000000000 +0200
+++ kdegraphics-4.4.5/libs/CMakeLists.txt	2012-07-24 02:19:51.227717070 +0200
@@ -1,5 +1,4 @@
 add_subdirectory(libkipi)
-add_subdirectory(libkdcraw)
 add_subdirectory(libkexiv2)
 if (WIN32 OR SANE_FOUND)
     add_subdirectory(libksane)
