diff -up kdebase-workspace-4.2.0/CMakeLists.txt.orig kdebase-workspace-4.2.0/CMakeLists.txt
--- kdebase-workspace-4.2.0/CMakeLists.txt.orig	2009-02-18 15:47:51.000000000 +0100
+++ kdebase-workspace-4.2.0/CMakeLists.txt	2009-02-18 15:50:34.000000000 +0100
@@ -59,9 +59,9 @@ endif(PYQT4_FOUND AND PYQT4_VERSION STRL
 
 macro_optional_find_package(PyKDE4)
 set(PythonSupport_FOUND FALSE)
-if(PYTHONLIBRARY_FOUND AND SIP_FOUND AND PYQT4_FOUND AND PYKDE4_FOUND)
+if(PYTHONLIBRARY_FOUND)
   set(PythonSupport_FOUND TRUE)
-endif(PYTHONLIBRARY_FOUND AND SIP_FOUND AND PYQT4_FOUND AND PYKDE4_FOUND)
+endif(PYTHONLIBRARY_FOUND)
 
 macro_log_feature(PythonSupport_FOUND "Python" "KDE Python support" "http://techbase.kde.org/Development/Languages/Python" FALSE "" "Needed to build plasma scriptengine for Python")
 #### End Python support ####
