diff -ru kde-baseapps-4.9.5.orig/CMakeLists.txt kde-baseapps-4.9.5/CMakeLists.txt
--- kde-baseapps-4.9.5.orig/CMakeLists.txt	2012-11-30 08:28:23.000000000 +0100
+++ kde-baseapps-4.9.5/CMakeLists.txt	2013-02-09 13:29:31.698421912 +0100
@@ -21,20 +21,13 @@
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/lib/konq ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
 
 add_subdirectory( lib )
-add_subdirectory( dolphin )
 add_subdirectory( kdialog )
-add_subdirectory( keditbookmarks )
-add_subdirectory( konqueror )
-add_subdirectory( kfind )
-add_subdirectory( plasma )
-add_subdirectory( konq-plugins )
 
 if ( Q_WS_MAC )
   add_subdirectory( kdepasswd )
 endif ( Q_WS_MAC )
 
 if ( Q_WS_X11 )
-  add_subdirectory( nsplugins )
   add_subdirectory( kdepasswd )
 endif ( Q_WS_X11 )
 macro_optional_add_subdirectory( doc )
diff -ru kde-baseapps-4.9.5.orig/doc/CMakeLists.txt kde-baseapps-4.9.5/doc/CMakeLists.txt
--- kde-baseapps-4.9.5.orig/doc/CMakeLists.txt	2012-11-30 08:28:23.000000000 +0100
+++ kde-baseapps-4.9.5/doc/CMakeLists.txt	2013-02-09 13:37:41.649388904 +0100
@@ -1,4 +1 @@
-add_subdirectory(dolphin)
-add_subdirectory(kfind)
-add_subdirectory(konqueror)
 add_subdirectory(kdepasswd)
