--- kdelibs-4.4.3/CMakeLists.txt.orig	2010-05-29 16:02:33.640549306 +0200
+++ kdelibs-4.4.3/CMakeLists.txt	2010-05-29 16:04:15.045180968 +0200
@@ -56,14 +56,6 @@ macro_log_feature(PERL_FOUND "Perl" "Per
 find_package(ZLIB)
 macro_log_feature(ZLIB_FOUND "ZLib" "The Zlib compression library" "http://www.zlib.net" TRUE "" "Required by the core KDE libraries and some critical kioslaves.")
 
-set(STRIGI_MIN_VERSION 0.6.3)
-find_package(Strigi REQUIRED)
-macro_log_feature(STRIGI_FOUND "Strigi" "A fast and small desktop searching program" "http://strigi.sourceforge.net" TRUE "${STRIGI_MIN_VERSION}" "Required by some critical kioslaves.")
-
-set(LIBATTICA_MIN_VERSION "0.1.1")
-find_package(LibAttica)
-macro_log_feature(LIBATTICA_FOUND "libattica" "A library to access Open Collaboration Service providers" "http://websvn.kde.org/trunk/kdesupport/attica/" TRUE "" "Required for Get Hot New Stuff.")
-
 #optional features
 if(X11_FOUND)
   #X11_Xrender discovery is done by FindX11
@@ -84,27 +76,9 @@ macro_log_feature(BZIP2_FOUND "BZip2" "A
 macro_optional_find_package(LibLZMA)
 macro_log_feature(LIBLZMA_FOUND "LZMA/XZ" "A very high compression ratio data compressor" "http://tukaani.org/xz/" FALSE "" "Provides the ability to read and write xz compressed data files.")
 
-macro_optional_find_package(OpenSSL)
-macro_log_feature(OPENSSL_FOUND "OpenSSL" "A toolkit implementing the Secure Sockets Layer (SSL v2/v3) and Transport Layer Security (TLS v1) protocols" "http://openssl.org" FALSE "" "STRONGLY RECOMMENDED: KDE uses OpenSSL for the bulk of secure communications, including secure web browsing via HTTPS.")
-
-macro_optional_find_package(Libintl)
-macro_log_feature(LIBINTL_FOUND "Libintl" "Library for message translation" "http://www.gnu.org/software/gettext" FALSE "" "STRONGLY RECOMMENDED: Enables KDE to be available in many different languages.")
-
 macro_optional_find_package(OpenGL)
 macro_log_feature(OPENGL_FOUND "OpenGL" "API for developing portable, interactive 2D and 3D graphics applications" "http://mesa3d.sourceforge.net" FALSE "" "STRONGLY RECOMMENDED: The 3D hardware acceleration available through the OpenGL API is used in applications ranging from graphics and modellers to screensavers and video players.")
 
-set(SOPRANO_MIN_VERSION "2.3.70")
-macro_optional_find_package(Soprano COMPONENTS PLUGIN_RAPTORPARSER PLUGIN_REDLANDBACKEND)
-macro_log_feature(SOPRANO_FOUND "Soprano" "Semantic Desktop Storing" "http://soprano.sourceforge.net" FALSE "${SOPRANO_MIN_VERSION}" "Provide metadata support (for semantic desktop).")
-macro_log_feature(SOPRANO_PLUGIN_RAPTORPARSER_FOUND "Soprano Raptor Parser" "RDF parser plugin for Soprano" "http://soprano.sourceforge.net" FALSE "" "The Soprano raptor parser plugin is required to build the Nepomuk semantic desktop system.")
-macro_log_feature(SOPRANO_PLUGIN_REDLANDBACKEND_FOUND "Soprano Redland Backend" "Redland storage backend for Soprano" "http://soprano.sourceforge.net" FALSE "" "The Soprano redland backend is required to build the Nepomuk semantic desktop system.")
-
-macro_optional_find_package(SharedDesktopOntologies)
-macro_log_feature(SHAREDDESKTOPONTOLOGIES_FOUND "Shared desktop ontologies" "Desktop ontologies" "http://oscaf.sourceforge.net" FALSE "" "Ontologies necessary for the Nepomuk semantic desktop.")
-
-macro_optional_find_package(QCA2)
-macro_log_feature(QCA2_FOUND "QCA2" "Qt Cryptographic Architecture" "http://delta.affinix.com/qca" FALSE "2.0.0" "Needed for the plasma remote widgets functionality.")
-
 ################# Disallow in-source build #################
 
 macro_ensure_out_of_source_build("kdelibs requires an out of source build. Please create a separate build directory and run 'cmake path_to_kdelibs [options]' there.")
@@ -251,9 +225,6 @@ add_subdirectory( kjsembed )
 add_subdirectory( kio )
 add_subdirectory( solid )
 add_subdirectory( kded       )
-if (QT_QT3SUPPORT_FOUND)
-   add_subdirectory( kde3support  )
-endif (QT_QT3SUPPORT_FOUND)
 add_subdirectory( kfile )
 add_subdirectory( kconf_update )
 add_subdirectory( kdoctools  )
@@ -277,7 +248,6 @@ add_subdirectory( kross )
 add_subdirectory( security )
 add_subdirectory( plasma )
 add_subdirectory( kunitconversion )
-add_subdirectory( kdewebkit )
 add_subdirectory( includes )
 
 macro_optional_add_subdirectory( doc  )
