diff -Naur kdelibs-4.1.72/kdecore/kernel/kstandarddirs.cpp kdelibs-4.1.72.kstandarddirs/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.1.72/kdecore/kernel/kstandarddirs.cpp	2008-11-12 18:31:44.000000000 +0100
+++ kdelibs-4.1.72.kstandarddirs/kdecore/kernel/kstandarddirs.cpp	2008-11-12 18:38:40.000000000 +0100
@@ -1002,7 +1002,9 @@
                  pit != prefixList->end();
                  ++pit)
             {
-	        if((*pit)!=installprefix||installdir.isEmpty())
+                // "exe" never has a custom install path, and the check triggers
+                // a false positive due to the libexecdir patch
+	        if((*pit)!=installprefix||installdir.isEmpty()||!strcmp("exe", type))
 	        {
                     for (QStringList::ConstIterator it = dirs.constBegin();
                          it != dirs.constEnd(); ++it)
@@ -1016,6 +1018,11 @@
                         if ((local || testdir.exists()) && !candidates.contains(path))
                             candidates.append(path);
                     }
+                    // special-case "config" (forward porting Chris Cheney's
+                    // hack) - we want /etc/kde after the local config paths
+                    // and before the ones in /usr (including kde-profile)
+                    if (local && !strcmp("config", type))
+                       candidates.append("/etc/kde/");
                     local = false;
                 }
 	        else
