Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/config/klipperrc
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/config/klipperrc
@@ -0,0 +1,2 @@
+[General]
+AutoStart=false
Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/applications/kde4/klipper.desktop
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/applications/kde4/klipper.desktop
@@ -0,0 +1,4 @@
+[Desktop Entry]
+NoDisplay=true
+Name=Removed entry
+Exec=/bin/true
