Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/applications/kde4/nepomukbackup.desktop
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/applications/kde4/nepomukbackup.desktop
@@ -0,0 +1,4 @@
+[Desktop Entry]
+NoDisplay=true
+Name=Removed entry
+Exec=/bin/true
Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/applications/kde4/nepomukcontroller.desktop
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/applications/kde4/nepomukcontroller.desktop
@@ -0,0 +1,4 @@
+[Desktop Entry]
+NoDisplay=true
+Name=Removed entry
+Exec=/bin/true
Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/autostart/nepomukcontroller.desktop
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/autostart/nepomukcontroller.desktop
@@ -0,0 +1,2 @@
+[Desktop Entry]
+Hidden=true
Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/autostart/nepomukserver.desktop
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/autostart/nepomukserver.desktop
@@ -0,0 +1,2 @@
+[Desktop Entry]
+Hidden=true
Index: kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/kde4/services/kcm_nepomuk.desktop
===================================================================
--- /dev/null
+++ kde-settings-4.9-19/usr/share/kde-settings/kde-profile/default/share/kde4/services/kcm_nepomuk.desktop
@@ -0,0 +1,2 @@
+[Desktop Entry]
+Hidden=true
Index: kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/config/nepomukserverrc
===================================================================
--- /dev/null
+++ kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/config/nepomukserverrc
@@ -0,0 +1,5 @@
+[Basic Settings]
+Start Nepomuk=false
+
+[Service-nepomukstrigiservice]
+autostart=false
