Index: kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/applications/Thunar.desktop
===================================================================
--- /dev/null
+++ kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/applications/Thunar.desktop
@@ -0,0 +1,4 @@
+[Desktop Entry]
+NoDisplay=true
+Name=Removed entry
+Exec=/bin/true
Index: kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/applications/thunar-settings.desktop
===================================================================
--- /dev/null
+++ kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/applications/thunar-settings.desktop
@@ -0,0 +1,4 @@
+[Desktop Entry]
+NoDisplay=true
+Name=Removed entry
+Exec=/bin/true
