diff -ru kde-settings.orig/etc/kde/kdm/kdmrc kde-settings/etc/kde/kdm/kdmrc
--- kde-settings.orig/etc/kde/kdm/kdmrc	2010-05-28 16:04:24.000000000 +0200
+++ kde-settings/etc/kde/kdm/kdmrc	2012-10-04 03:00:56.225261762 +0200
@@ -453,7 +453,7 @@
 #ServerCmd=/usr/X11R6/bin/X
 # Additional arguments for the X-servers for local sessions.
 # This string is subject to word splitting. Default is ""
-ServerArgsLocal=-nr -nolisten tcp
+ServerArgsLocal=-background none -nolisten tcp
 # Additional arguments for the X-servers for remote sessions.
 # This string is subject to word splitting. Default is ""
 #ServerArgsRemote=
