Index: kde-settings-4.9-19/etc/kde/kdm/kdmrc
===================================================================
--- kde-settings-4.9-19.orig/etc/kde/kdm/kdmrc
+++ kde-settings-4.9-19/etc/kde/kdm/kdmrc
@@ -57,7 +57,7 @@ ConfigVersion=2.3
 StaticServers=:0
 # List of on-demand displays. See StaticServers for syntax.
 # Default is ""
-ReserveServers=:1,:2,:3
+ReserveServers=
 # VTs to allocate to X-servers. A negative number means that the VT will be
 # used only if it is free. If all VTs in this list are used up, the next free
 # one greater than the last one in this list will be allocated.
