Index: kde-settings-4.9-19/etc/kde/kdm/kdmrc
===================================================================
--- kde-settings-4.9-19.orig/etc/kde/kdm/kdmrc
+++ kde-settings-4.9-19/etc/kde/kdm/kdmrc
@@ -294,7 +294,7 @@ GUIStyle=Plastique
 LogoArea=Logo
 # The image to show when LogoArea=Logo.
 # Default is ""
-LogoPixmap=/usr/share/icons/hicolor/96x96/apps/fedora-logo-icon.png
+LogoPixmap=/usr/share/icons/hicolor/96x96/apps/qubes-logo-icon.png
 # The relative coordinates (X,Y in percent) of the center of the greeter.
 # Default is "50,50"
 #GreeterPos=30,40
@@ -313,7 +313,7 @@ LogoPixmap=/usr/share/icons/hicolor/96x9
 # - %m -> the machine (hardware) type
 # - %% -> a single %
 # Default is "Welcome to %s at %n"
-GreetString=Fedora 18 (Spherical Cow)
+GreetString=Qubes OS
 # The font for the greeter headline.
 # Default is "charter,24,bold"
 GreetFont=Sans,16,-1,5,75,0,0,0,0,0
@@ -442,7 +442,7 @@ ForgingSeed=1108476160
 UseTheme=true
 # The theme to use for the greeter. Can point to either a directory or an XML
 # file. Default is "/usr/share/kde4/apps/kdm/themes/circles"
-Theme=/usr/share/kde4/apps/kdm/themes/SphericalCow
+Theme=/usr/share/kde4/apps/kdm/themes/oxygen
 # Set the widget color scheme 
 #ColorScheme=ObsidianCoast
 
