Index: kde-settings-23-11/etc/kde/kdm/kdmrc
===================================================================
--- kde-settings-23-11.orig/etc/kde/kdm/kdmrc
+++ kde-settings-23-11/etc/kde/kdm/kdmrc
@@ -322,7 +322,7 @@ GUIStyle=Plastique
 LogoArea=Logo
 # The image to show when LogoArea=Logo.
 # Default is ""
-LogoPixmap=/usr/share/icons/hicolor/96x96/apps/fedora-logo-icon.png
+LogoPixmap=/usr/share/icons/hicolor/96x96/apps/qubes-logo-icon.png
 # The relative coordinates (X,Y in percent) of the center of the greeter.
 # Default is "50,50"
 #GreeterPos=30,40
@@ -341,7 +341,7 @@ LogoPixmap=/usr/share/icons/hicolor/96x9
 # - %m -> the machine (hardware) type
 # - %% -> a single %
 # Default is "Welcome to %s at %n"
-GreetString=Fedora 22 (Twenty Two)
+GreetString=Qubes OS
 # Whether the fonts used in the greeter should be antialiased.
 # Default is false
 AntiAliasing=true
@@ -477,7 +477,7 @@ UseTheme=true
 # The theme to use for the greeter. Can point to either a directory or an XML
 # file.
 # Default is ""
-Theme=/usr/share/kde4/apps/kdm/themes/F23
+Theme=/usr/share/kde4/apps/kdm/themes/oxygen
 # Enable the Alt-Ctrl-D shortcut to toggle greeter theme debugging.
 # Default is false
 #AllowThemeDebug=true
