--- kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/plasma/plasmoids/org.kde.plasma.kickoff/contents/ui/FullRepresentation.qml	2016-04-19 13:15:58.000000000 +0200
+++ kde-settings-23-11/usr/share/kde-settings/kde-profile/default/share/plasma/plasmoids/org.kde.plasma.kickoff/contents/ui/FullRepresentation.qml	2016-05-14 02:26:47.830000000 +0200
@@ -103,11 +103,13 @@
                 when: mainTabGroup.currentTab == applicationsPage
                 source: Qt.resolvedUrl("ApplicationsView.qml")
             }
+            /* remove in Qubes OS
             PlasmaExtras.ConditionalLoader {
                 id: systemPage
                 when: mainTabGroup.currentTab == systemPage
                 source: Qt.resolvedUrl("ComputerView.qml")
             }
+            */
             PlasmaExtras.ConditionalLoader {
                 id: recentlyUsedPage
                 when: mainTabGroup.currentTab == recentlyUsedPage
@@ -367,12 +369,14 @@
             iconSource: "applications-other"
             text: i18n("Applications")
         }
+        /* remove in Qubes OS
         KickoffButton {
             id: computerButton
             tab: systemPage
             iconSource: pmSource.data["PowerDevil"] && pmSource.data["PowerDevil"]["Is Lid Present"] ? "computer-laptop" : "computer"
             text: i18n("Computer")
         }
+        */
         KickoffButton {
             id: usedButton
             tab: recentlyUsedPage
