--- a/usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kicker.js
+++ b/usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kicker.js
@@ -22,7 +22,8 @@
    Copyright Canonical Ltd, may be copied under the GNU GPL 2 or later
 */
 
-if ( applet.readConfig("icon", "start-here-kde") == "start-here-kde" ) {
+if ( applet.readConfig("icon", "start-here-kde") == "start-here-kde" || 
+     applet.readConfig("icon", "start-here-kde") == "start-here") {
   applet.currentConfigGroup = ["General"]
-  applet.writeConfig("icon", "start-here");
+  applet.writeConfig("icon", "qubes-logo-icon");
 }
--- a/usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kickerdash.js
+++ b/usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kickerdash.js
@@ -22,7 +22,8 @@
    Copyright Canonical Ltd, may be copied under the GNU GPL 2 or later
 */
 
-if ( applet.readConfig("icon", "start-here-kde") == "start-here-kde" ) {
+if ( applet.readConfig("icon", "start-here-kde") == "start-here-kde" || 
+     applet.readConfig("icon", "start-here-kde") == "start-here") {
   applet.currentConfigGroup = ["General"]
-  applet.writeConfig("icon", "start-here");
+  applet.writeConfig("icon", "qubes-logo-icon");
 }
--- a/usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kickoff.js
+++ b/usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kickoff.js
@@ -22,7 +22,8 @@
    Copyright Canonical Ltd, may be copied under the GNU GPL 2 or later
 */
 
-if ( applet.readConfig("icon", "start-here-kde") == "start-here-kde" ) {
+if ( applet.readConfig("icon", "start-here-kde") == "start-here-kde" || 
+     applet.readConfig("icon", "start-here-kde") == "start-here") {
   applet.currentConfigGroup = ["General"]
-  applet.writeConfig("icon", "start-here");
+  applet.writeConfig("icon", "qubes-logo-icon");
 }
